### PR TITLE
[Hub Generated] Review request for Microsoft.OperationalInsights to add version stable/2022-10-01

### PR DESCRIPTION
**Pull request by dulikvor**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/dulikvor-operationalinsights-Microsoft.OperationalInsights-2022-10-01...main/).
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [ ] New API version. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by dulikvor** [comment]
Breaking changes were already approved in the following PR:
https://github.com/Azure/azure-rest-api-specs/pull/26096

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:70]: >v2 [](http://example.com/codeflow?start=71&length=2)

please use v3 or newer

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json:710]: -1 is not between 4, so how can the minimum be 4? And what does 'default to table retention' mean?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/examples/ClustersCreate.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/examples/ClustersCreate.json:31]: Would it make sense to just give the secrets uri, instead of all the parts like this? It seems a bit more internetty hypertexty.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:622]: Have you looked at the possibility of reusing definitions from common-types types.json or managedidentity.json insted?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:639]: >esourceId id [](http://example.com/codeflow?start=31&length=12)

typo?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:634]: > name id the assigned worksp [](http://example.com/codeflow?start=29&length=28)

typo?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:642]: >associateDate [](http://example.com/codeflow?start=9&length=13)

naming guidelines for timestamps are to use At suffix, like 'associatedAt' or 'createdAt'.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:637]: >resourceId [](http://example.com/codeflow?start=9&length=10)

"format": "arm-id" right?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:535]: >GB [](http://example.com/codeflow?start=60&length=2)

Nit consider being explicit about Gigabytes and not gibibytes.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:376]: >keyVaultProperties [](http://example.com/codeflow?start=9&length=18)

is it readonly?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:513]: are keyvault properties readonly?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/examples/ClustersSubscriptionList.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/examples/ClustersSubscriptionList.json:30]: >TestResourceLock [](http://example.com/codeflow?start=21&length=16)

I wonder why it is called this

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (TimLovellSmith) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:513]: I guess not, since there's an update sample for them. You might want to include them in a create sample also then.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json:710]: -1 is a special value which indicate - Delete current value.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:642]: Out of scope of change and a breaking change.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:513]: They are not read-only, values are provided explicitly by customers.

Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:513]: Keyvault usage is a niche, i prefer to provide the common case scenario in create example.
User assigned identity and key vault usage are covered in Update.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/examples/ClustersSubscriptionList.json
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/examples/ClustersSubscriptionList.json:30]: No actual meaning, changed.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json:376]: Its not.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (sjanamma) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json]: This is good to have. Please consider this change for the next version atleast

**Comment by sjanamma** [comment]
PR has breaking changes that have not been approved yet. Please get a breaking changes approval from Azure Breaking Changes Reviewers <azbreakchangereview@microsoft.com> and ARM can review the changes afterwards. Once you have the approval, please remove the "ARMChangesRequested" label from the PR to make the PR visible to the ARM reviewers.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json]: Change was done. please review latest version.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json]: Done.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json]: Done.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Clusters.json]: Done.

**Comment by dulikvor** [comment]
@sjanamma breaking change was approved, you can review it here, same PR, we just re opened it using Open API tools to make sure the base version is listed:
https://github.com/Azure/azure-rest-api-specs/pull/26096

**Comment by mikekistler** [comment]
Breaking changes previously reviewed and approved in https://github.com/Azure/azure-rest-api-specs/pull/26096

**Comment by visingla-ms** [comment]
Please address this comment and get SDK breaking changes reviewed from SDK reviewers. Rest LGTM.

**Comment by Alancere** [comment]
Hi @dulikvor, please fix `SDK azure-sdk-for-go` CI.

**Comment by dulikvor** [comment]
> Hi @dulikvor, please fix `SDK azure-sdk-for-go` CI.

Go was failing due to multiple versions of common types, i had to revert our recent change to upgrade some of them from V2 to V5, also change is out of scope for other resources, now all looks good. your assistance with pushing this PR will be appreciated.

**Review by ms-zhenhua — APPROVED** [review]

**Pull request merged**
